torchvision/transforms/functional_tensor.py

```diff
@@ -327,3 +327,63 @@ def _hsv2rgb(img):
     a4 = torch.stack((a1, a2, a3))
 
     return torch.einsum("ijk, xijk -> xjk", mask.to(dtype=img.dtype), a4)
+
+
+def pad(img: Tensor, padding: List[int], fill: int, padding_mode: str = "constant") -> Tensor:
+    r"""Pad the given Tensor Image on all sides with specified padding mode and fill value.
+
+    Args:
+        img (Tensor): Image to be padded.
+        padding (int or tuple or list): Padding on each border. If a single int is provided this
```

Comment:
doc should be updated, as we only support `List[int]` in torchscript

Reply:
Actually, we can support both: tuple and list as tuple is casted to list below (L361). Annotation is kept as `List` for torchscript to pass. In the tests `test_pad` I check both types: `[4, 4]` and `(2, 2, 2, 2)`.
Do you want me add some note about that ?

Reply:
yeah, I think we should be careful with how we describe our support.
If we say we support `tuple` but it fails in torchscript, that's a bad user experience.

I would say it might be better to say we only support `List`, and as a bonus we support `Tuple` in non-torchscript mode.

Thoughts?

Reply:
Yes, we need to provide correct information about what is supported and when. In case of pad in torchscript or non-torchscipt mode, both types (tuple, list) are supported :
```python
>>> from torchvision.transforms.functional_tensor import pad
>>> import torch
>>> tensor = torch.randint(0, 255, (3, 10, 12), dtype=torch.uint8)
>>> r1 = pad(tensor, (2, 2), fill=0, padding_mode="constant")
>>> r2 = pad(tensor, [2, 2], fill=0, padding_mode="constant")
>>> r1.eq(r2).all()
tensor(True)
>>> script_pad = torch.jit.script(pad)
>>> r1 = script_pad(tensor, (2, 2), fill=0, padding_mode="constant")
>>> r2 = script_pad(tensor, [2, 2], fill=0, padding_mode="constant")
>>> r1.eq(r2).all()
tensor(True)
```

Anyway, I'll update the docstring to make it crystal clear :)